maestro-cli/src/main/java/maestro/cli/insights/TestAnalysisManager.kt

```diff
@@ -0,0 +1,179 @@
+                    commands.add(AnalysisLog(data = data, path = path))
+                }
+
+                fileName == "maestro.log" -> {
+                    logs.add(AnalysisLog(data = data, path = path))
+                }
+            }
+        }
+
+        val filteredScreenshots = filterSimilarScreenshots(screenshots)
+
+        return AnalysisDebugFiles(
+            logs = logs,
+            commands = commands,
+            screenshots = filteredScreenshots,
+        )
+    }
+
+    private val screenshotsDifferenceThreshold = 5.0
+
+    private fun filterSimilarScreenshots(
+        screenshots: List<AnalysisScreenshot>
+    ): List<AnalysisScreenshot> {
+        val uniqueScreenshots = mutableListOf<AnalysisScreenshot>()
+
+        for (screenshot in screenshots) {
+            val isSimilar = uniqueScreenshots.any { existingScreenshot ->
+                val diffPercent = ImageComparisonUtil.getDifferencePercent(
+                    ImageComparisonUtil.readImageFromResources(existingScreenshot.path.toString()),
+                    ImageComparisonUtil.readImageFromResources(screenshot.path.toString())
+                )
+                diffPercent <= screenshotsDifferenceThreshold
+            }
+
+            if (!isSimilar) {
+                uniqueScreenshots.add(screenshot)
+            }
+        }
+
+        return uniqueScreenshots
+    }
+
+    /**
+     * The Notification system for Test Analysis.
+     *  - Uses configuration from $XDG_CONFIG_HOME/maestro/analyze-notification.json.
+     */
+    companion object AnalysisNotification {
+        private const val DISABLE_NOTIFICATION_ENV_VAR = "MAESTRO_CLI_ANALYSIS_NOTIFICATION_DISABLED"
+        private val disabled: Boolean
+            get() = System.getenv(DISABLE_NOTIFICATION_ENV_VAR) == "true"
+
+        private val notificationStatePath: Path = EnvUtils.xdgStateHome().resolve("analyze-notification.json")
+
+        private val JSON = jacksonObjectMapper().apply {
+            registerModule(JavaTimeModule())
+            enable(SerializationFeature.INDENT_OUTPUT)
+        }
+
+        private val shouldNotNotify: Boolean
+            get() = disabled || notificationStatePath.exists() && notificationState.acknowledged
+
+        private val notificationState: AnalysisNotificationState
+            get() = JSON.readValue<AnalysisNotificationState>(notificationStatePath.readText())
+
+        fun maybeNotify() {
+            if (shouldNotNotify) return
+
+            println(
+                listOf(
+                    "Try out our new Analyze with Ai feature.\n",
+                    "See what's new:",
+                    "> https://maestro.mobile.dev/cli/test-suites-and-reports#analyze",
+                    "Analyze command:",
+                    "$ maestro test flow-file.yaml --analyze | bash\n",
+                    "To disable this notification, set $DISABLE_NOTIFICATION_ENV_VAR environment variable to \"true\" before running Maestro."
```

Comment:
the correct env var name is $MAESTRO_CLI_ANALYSIS_NOTIFICATION_DISABLED, not $DISABLE_NOTIFICATION_ENV_VAR